lib/ansible/modules/cloud/vmware/vmware_guest_disk.py

```diff
@@ -482,6 +482,10 @@ def ensure_disks(self, vm_obj=None):
                     disk_spec.device.backing.thinProvisioned = True
                 elif disk['disk_type'] == 'eagerzeroedthick':
                     disk_spec.device.backing.eagerlyScrub = True
+                # get Storage DRS recommended datastore from the datastore cluster
+                if disk['datastore_cluster'] is not None:
```

Comment:
I didn't realize this is a new key. sorry for that @rrotaru. You need to document it here https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_guest_disk.py#L79-L115

In addition, it would be nice to make us of the new key in the functional tests: https://github.com/ansible/ansible/blob/devel/test/integration/targets/vmware_guest_disk/tasks/main.yml
Finally, you should also add a changelog fragment ( https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#changelogs ) to document the change.

You can find some documentation about the function tests here: https://docs.ansible.com/ansible/devel/dev_guide/platforms/vmware_guidelines.html

Don't hesitate to ping me if you need help.

Reply:
Ah, I did not realize I was exposing a new key here. Let me re-work some of the implementation. I would prefer to keep `datastore` as the key to be consistent with the rest of the modules under `vmware` and not add any new attributes to the module. I'm really only looking into doing a bugfix here, not a feature enhancement.

I was using that key to determine whether or not we need to make a call to `get_recommended_datastore`  during the `ensure_disks` method, but I think there is a better way to solve this. Great catch, thanks!

Reply:
There are a couple of attributes in the resulting list from `sanitize_disk_inputs` that do not line up with the keys available from the documentation:

https://github.com/rrotaru/ansible/blob/7f8d1aef1329f714ca7e89bf730fc08dd7c2c40d/lib/ansible/modules/cloud/vmware/vmware_guest_disk.py#L561-L570
`disk_index`, `destroy`, `filename`, `disk_unit_number`, `scsi_controller`, and `disk_mode` are all defined as "keys" for this method but not available in the documentation.

As the logic stands today, we grab `datastore_cluster` from here:
https://github.com/rrotaru/ansible/blob/7f8d1aef1329f714ca7e89bf730fc08dd7c2c40d/lib/ansible/modules/cloud/vmware/vmware_guest_disk.py#L591

And assign it to the resulting array here:
https://github.com/rrotaru/ansible/blob/7f8d1aef1329f714ca7e89bf730fc08dd7c2c40d/lib/ansible/modules/cloud/vmware/vmware_guest_disk.py#L597-L599

Thus the user is 
a) not expected to pass in `datastore_cluster` as a key (functionally it works by taking `datastore` as the existing key and determining if it is an instance of vim.StoragePod for us)
b) not able to pass in the key `datastore_cluster` and have it recognized by the module, since we do not copy all of the attributes from `self.desired_disks` (e.g. we never check or assign or copy `disk['datastore_cluster']` in that array, which would be the "new key" you are mentioning).

@goneri I do not believe this is a new key that would require documentation.
I will still add the changelog fragment under bugfix